pandas/_libs/algos_groupby_helper.pxi.in

```diff
@@ -55,37 +56,65 @@ def group_add_{{name}}(ndarray[{{dest_type2}}, ndim=2] out,
 
     with nogil:
 
-        if K > 1:
+        if skipna == False:
+            if K > 1:
 
-            for i in range(N):
-                lab = labels[i]
-                if lab < 0:
-                    continue
-
-                counts[lab] += 1
-                for j in range(K):
-                    val = values[i, j]
+                for i in range(N):
```

Comment:
I am running into a problem that I am not that sure about. The support of datetimelikes in the cython routines is done. So for example, `sum` now works on`timedelta` values just like it's non-cython implementation. 

The problem is it's raising `AssertionError("Gaps in blk ref_locs")` in `_rebuild_blknos_and_blklocs` while initializing a `BlockManager` in `mgr = BlockManager(blocks, [items, index])` in the method `self._wrap_agged_blocks(new_items, new_blocks)` after the cython routine is done and we have a timedelta block.

Can you please guide me on this issue? I am at a loss this time.